phoenix-core/src/it/java/org/apache/phoenix/end2end/ParameterizedAlterTableCascadeIT.java

```diff
@@ -0,0 +1,217 @@
+        ps.executeUpdate();
+        ps = conn.prepareStatement("UPSERT INTO us_population VALUES('AZ','Phoenix',1461575)");
+        ps.executeUpdate();
+        ps = conn.prepareStatement("UPSERT INTO us_population VALUES('TX','San Antonio',1256509)");
+        ps.executeUpdate();
+        ps = conn.prepareStatement("UPSERT INTO us_population VALUES('CA','San Diego',1255540)");
+        ps.executeUpdate();
+        ps = conn.prepareStatement("UPSERT INTO us_population VALUES('TX','Dallas',1213825)");
+        ps.executeUpdate();
+        ps = conn.prepareStatement("UPSERT INTO us_population VALUES('CA','San Jose',912332)");
+        ps.executeUpdate();
+    }
+
+    // Test with ALTER TABLE CASCADE INDEX ALL
+    @Test
+    public void testAlterDBOAddCascadeIndexAll() throws SQLException {
+
+
+        String query = "ALTER "+ databaseObject + " " + databaseObjectName +" ADD new_column VARCHAR CASCADE INDEX ALL";
+        conn.createStatement().execute(query);
+        query = "EXPLAIN SELECT new_column FROM "+databaseObjectName;
+        ResultSet rs = conn.prepareStatement(query).executeQuery();
+        rs.next();
+        //confirm it is using global index to access the column
+        Assert.assertTrue(rs.getString(1).contains(planOutput.toUpperCase()));
+    }
+
+    // Test with ALTER VIEW CASCADE INDEX ALL
+    @Test
+    public void testAlterDBOAddCascadeIndexAllUpsert() throws SQLException {
+        String query = "ALTER "+ databaseObject + " " + databaseObjectName +" ADD new_column_3 VARCHAR CASCADE INDEX ALL";
+        conn.createStatement().execute(query);
+        PreparedStatement ps;
+        if(isViewIndex) {
+            ps = conn.prepareStatement("UPSERT INTO us_population_gv(state,city,population,city_area,avg_fam_size,new_column_3) " +
+                    "VALUES('CA','Santa Barbara',912332,1300,4,'test_column')");
+        } else {
+            ps = conn.prepareStatement("UPSERT INTO us_population(state,city,population,new_column_3) " +
+                    "VALUES('CA','Santa Barbara',912332,'test_column')");
+        }
+        ps.executeUpdate();
+        query = "EXPLAIN SELECT new_column_3 FROM "+databaseObjectName+" where new_column_3 = 'test_column'";
+        ResultSet rs = conn.prepareStatement(query).executeQuery();
+        rs.next();
+        //confirm it is using global index to access the column
+        Assert.assertTrue(rs.getString(1).contains(planOutput.toUpperCase()));
+        query = "SELECT new_column_3 FROM "+databaseObjectName+" where new_column_3 = 'test_column'";
+        rs = conn.prepareStatement(query).executeQuery();
+        rs.next();
+        Assert.assertEquals(rs.getString(1),"test_column");
+    }
+
+
+
+    // Test with CASCADE INDEX <index_name>
+    @Test
+    public void testAlterDBOAddCascadeIndex() throws SQLException {
+        String query = "ALTER "+databaseObject+" "+databaseObjectName + " ADD new_column_1 FLOAT CASCADE INDEX "+indexesName.split(",")[0];
```

Comment:
Can you add multiple columns? If yes, can they all cascade? If yes, let's add a test for that.